### PR TITLE
Add Generative Prompts section

### PR DESCRIPTION
**Pull request by pwoznic**
Added new prompts section including topic about customizing prompts with metadata.

### CODE REVIEW

File: www/docs/prompts/custom_prompts_with_metadata.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+| $vectaraQuery  | The query provided by the user  | Generate a summary in $vectaraOutChars characters in language '${vectaraLangCode}' for the query \"$esc.java(${vectaraQuery})\" solely based on the search results in this chat.  | Generate a summary in 512 characters in language 'ara' for the query \"Give me \"some\" search results.\" solely based on the search results in this chat.  |
+| $vectaraIdxWord  | A utility array to convert the index to words i.e "first", "second", "third", "forth", "fifth", "sixth", "seventh", "eighth", "ninth", "tenth"  | $vectaraIdxWord[0]  | first  |
+| $vectaraLangName  | Set to the requested language name. The language can either be requested explicitly or detected from the language of the query.  | You are a helpful assistant. Answer in ${vectaraLangName}.  | You are a helpful assistant. Answer in Arabic.  |
+| $vectaraQueryResults  | An array of query results is found in the response, sorted by relevance score.  | You are a helpful assistant. Answer in ${vectaraLangName}.  | {"role": "user", "content": "Give me the second search result."},{"role": "assistant", "content": "2nd result" },  |
```
Comment (eskibars) [inline www/docs/prompts/custom_prompts_with_metadata.md:29]: @pwoznic the Usage Input and Usage Output are not correct here.
The example input would be something like: `#foreach $qResult in $vectaraQueryResults`

The example output would be something like:`[ result_array ]`, but it's a bit difficult to show the result array in a meaningful way beyond that.

File: www/docs/prompts/custom_prompts_with_metadata.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+## Example Custom Prompt for an RFI Answering Bot
+
+The following example prompt creates a Request for information (RFI) 
+answering bot that includes metadata. First, we ask the generative LLM to 
+answer an RFI question and tell it how the results will come back from the 
+query. 
+
+We want to iterate through `$vectaraQueryResults` inserting the results 
+in the order that we like. `qResult.getText()` provides the most relelvant
```
Comment (eskibars) [inline www/docs/prompts/custom_prompts_with_metadata.md:71]: `$qResult` instead of `qResult`

**Review by eskibars — APPROVED** [review]
A few small changes to address; I've left suggestions.  Go ahead and merge after those are made

File: www/docs/prompts/custom_prompts_with_metadata.md (hunk 2 above)
Comment (pwoznic) [reply www/docs/prompts/custom_prompts_with_metadata.md]: Updated

File: www/docs/prompts/custom_prompts_with_metadata.md (hunk 1 above)
Comment (pwoznic) [reply www/docs/prompts/custom_prompts_with_metadata.md]: Updated example

**Pull request merged**